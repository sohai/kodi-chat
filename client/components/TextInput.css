.root {
  height: 35px;
}
.input {
  display: block;
  width: 100%;
  box-sizing: border-box;
  padding: 10px;
  border: 0;
}
.input:focus {
  outline: none;
}
