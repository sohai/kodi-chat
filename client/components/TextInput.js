import React from 'react';
import PropTypes from 'prop-types';
import styles from './TextInput.css';

const propTypes =
  PropTypes &&
  {
    //classes: PropTypes.object.isRequired
  };
const defaultProps = {};

export const TextInputComponent = ({ value, onChange, onEnter }) => (
  <div className={styles.root}>
    <input className={styles.input} value={value} onChange={onChange} onKeyPress={onEnter} />
  </div>
);

TextInputComponent.displayName = 'TextInput';
TextInputComponent.propTypes = propTypes;
TextInputComponent.defaultProps = defaultProps;
export default TextInputComponent;
