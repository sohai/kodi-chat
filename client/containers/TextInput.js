import React from 'react';
import { compose, pure, withState, withHandlers } from 'recompact';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { TextInput } from '../components/TextInput';
import proxy from '../providers/ws';
import messagesHandler from '../handlers/messages';

const propTypes = PropTypes && {};

export const TextInputRender = ({ value, onChange, onKeyPress }) => (
  <TextInput value={value} onChange={onChange} onKeyPress={onKeyPress} />
);

export const mapStateToProps = state => ({
  username: state.users.you.name
});

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      addMessage: messagesHandler.actions.addMessage
    },
    dispatch
  );

TextInputRender.propTypes = propTypes;
TextInputRender.displayName = 'TextInput';
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withState('value', 'updateValue', ''),
  withHandlers({
    onChange: props => event => {
      props.updateValue(event.target.value);
    },
    onKeyPress: props => event => {
      if (event.key === 'Enter') {
        const { value, username, updateValue } = props;
        props.addMessage({
          username,
          message: value
        });
        proxy.emit('new message', value);
        updateValue('');
      }
    }
  }),
  pure
)(TextInputRender);
