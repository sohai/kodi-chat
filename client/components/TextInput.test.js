/* global shallow, describe, it, expect */
/* eslint-disable import/first */
import React from 'react';
import TextInput from './TextInput';

describe('(Component) TextInput', () => {
  it('should render', () => {
    const props = {
      children: <span>test</span>
    };
    const wrapper = shallow(<TextInput {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
