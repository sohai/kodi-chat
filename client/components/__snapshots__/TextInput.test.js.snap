// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) TextInput should render 1`] = `
<div
  class="root"
>
  <input
    class="input"
  />
</div>
`;
